### PR TITLE
Inprogress message fix

### LINKED ISSUE #15735: Allow SiP "inProgress" messaging to be disabled
### User story(ies)

As a VFS developer, I want a way to disable the "in progress" message on the save-in-progress intro component, so that we don't have redundant text when we don't have any alternative in progress messaging to display.

image

### Additional details/notes/links

- https://github.com/department-of-veterans-affairs/vets-website/pull/14697

### Acceptance criteria

- [ ] "inProgress" message can be disabled by setting it to be an empty string

### Point(s) of contact

- @bkjohnson 

### What type of issue is this? 
- [x] Bug
- [ ] Request new feature
- [ ] Request update to existing feature
- [ ] Documentation change
- [ ] Other (please describe)

### PR DESCRIPTION
**Pull request by bkjohnson**
### Fixes department-of-veterans-affairs/va.gov-team/issues/15735

This allows the "in progress" part of the save-in-progress messaging to be disabled. This can be done by setting `formConfig.saveInProgress.messages.inProgress` to the empty string.

## Testing done

Unit tests added

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/forms-system/test/js/utilities/save-in-progress-messages.unit.spec.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { expect } from 'chai';
+
+import { inProgressMessage } from '../../../src/js/utilities/save-in-progress-messages';
+
+describe('inProgressMessage()', () => {
```
Comment (cvalarida) [inline src/platform/forms-system/test/js/utilities/save-in-progress-messages.unit.spec.js:5]: Oh dang. I could have sworn I wrote tests for this already. :confused: Must have been in a previous iteration that I deleted or never committed. Well, thanks for writing these! :raised_hands:

File: src/platform/forms-system/test/js/utilities/save-in-progress-messages.unit.spec.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    expect(inProgressMessage(formConfig)).to.eql(
+      'Your application is in progress.',
+    );
+  });
+
+  it('should return a message from the formConfig when properly set', () => {
+    const formConfig = {
+      saveInProgress: {
+        messages: {
+          inProgress: 'Bippity boppity boo!',
```
Comment (cvalarida) [inline src/platform/forms-system/test/js/utilities/save-in-progress-messages.unit.spec.js:22]: :joy:

File: src/platform/forms/save-in-progress/SaveInProgressIntro.jsx [hunk 3]
```diff
@@ -67,10 +68,14 @@ class SaveInProgressIntro extends React.Component {
                   <strong>Your {appType} is in progress</strong>
                 </div>
                 <div className="saved-form-metadata-container">
-                  <span className="saved-form-item-metadata">
-                    {inProgressMessage(formConfig)}
-                  </span>
-                  <br />
+                  {inProgressMessage && (
```
Comment (cvalarida) [inline src/platform/forms/save-in-progress/SaveInProgressIntro.jsx:71]: This is great. :+1:

**Review by cvalarida — APPROVED** [review]
This all looks good. Solves the problem at hand.

I'm thinking now that we should add the same functionality to the other SiP messages—`expiredMessage()` and `savedMessage()`. Could be confusing that we can set `inProgress` to `''` and have that not render the default, but if we pass an empty string to the others, it'll use the default messages.

What do you think? We can write a ticket for that and tackle them separate from this PR, since we're tracking this particular work as support right now.

**Comment by bkjohnson** [comment]
I agree - those other message helpers should behave the same. The only reason I didn't include those changes here was to keep the scope down.

**Pull request merged**